test/models/publisher_test.rb

```diff
@@ -17,6 +17,42 @@ class PublisherTest < ActiveSupport::TestCase
     Rails.application.secrets[:api_eyeshade_offline] = @prev_offline
   end
 
+  describe "#confirm_pending_email!" do
+    let(:publisher) { publishers(:default) }
+    let(:confirm_email) { nil }
+
+    describe "confirm_email" do
+      describe "pending_email" do
+        describe "!email" do
+          it "should set email to pending_email" do
+          end
+        end
+
+        describe "confirm_email == pending_email" do
+          it "should set email to pending_email" do
```

Comment:
Only set the email if pending_email is == to the confirm email HTTP parameter.
This is an extra safeguard against attackers that got hold of the auth token, without knowing the email address.
It's also kinda unneeded since we use UUIDs. It works as an extra security layer.